packages/aws-cdk-lib/aws-eks/lib/kubectl-handler/helm/__init__.py

```diff
@@ -110,11 +115,16 @@ def get_oci_cmd(repository, version):
     elif public_registry['registry'] is not None:
         logger.info("Found AWS public repository, will use default region as deployment")
         region = os.environ.get('AWS_REGION', 'us-east-1')
-
-        cmnd = [
-            f"aws ecr-public get-login-password --region us-east-1 | " \
-            f"helm registry login --username AWS --password-stdin {public_registry['registry']}; helm pull {repository} --version {version} --untar"
-            ]
+        
+        if is_ecr_public_available(region):
```

Comment:
I'm not sure I understand what is happening here. From what I can follow this code would only be executed if the repository that is passed in is a public repository,  in which case just running `helm pull` wouldn't work right?

Reply:
This means now that we are not in `aws` partition and probably in `aws-cn` partition but I need to helm install a chart located in ECR Public in us-east-1, which means:

1. I will need to helm pull that image straight from us-east-1 in `unauthenticated` way because I can't authenticate it in non `aws` partitions.
2. I can't run [helm registry login](https://helm.sh/docs/helm/helm_registry_login/) because I need to pull that image as unauthenticated.

Reply:
ahh ok thanks for the clarification.